utils: iio_info: do not print debug attribute values

dlech opened the pull request:
## PR Description

Do not print the values of debug attributes in iio_info.

It has come up multiple times that reading debug attributes can cause unintended side effects. For example if the IIO driver is currently doing a buffered read, reading the debug attribute could cause the chip to exit data capture mode and break the buffered read. Also, some registers could clear or reset something, like an interrupt, when a status register is read.

So general-purpose tools like iio_info should not read debug attributes that the user did not explicitly request.

Closes: https://github.com/analogdevicesinc/libiio/issues/1304
Link: https://lore.kernel.org/linux-iio/20250621170824.249c6b0c@jic23-huawei/

## PR Type
- [x] Bug fix (a change that fixes an issue)
- [ ] New feature (a change that adds new functionality)
- [ ] Breaking change (a change that affects other repos or cause CIs to fail)

## PR Checklist
- [x] I have conducted a self-review of my own code changes
- [x] I have commented new code, particulary complex or unclear areas
- [ ] I have checked that I did not intoduced new warnings or errors (CI output)
- [ ] I have checked that components that use libiio did not get broken
- [ ] I have updated the documentation accordingly (GitHub Pages, READMEs, etc)

dlech commented:
Umm... I think the static checker needs work. This is clearly not true.

![image](https://github.com/user-attachments/assets/c1f5bf1a-7762-423c-8357-effc85b1cbfd)

gastmaier commented:
Note: merging to main is merging to >= v1.
@dNechita if this is approved by you, should a similar pr be created for v0.x?

rgetz commented:
I would rather add a '-d' option, which enables/suppresses reading them - so people have a way to do what they have been doing for 10+ years.

nunojsa commented:
> I would rather add a '-d' option, which enables/suppresses reading them - so people have a way to do what they have been doing for 10+ years.

I tend to agree with the above. Arguably this should never have been added but it was and users have been using it. Even wonder what the default should be? Meaning, should we keep behavior and add a switch to shut it down? But given that this can lead to unexpected bugs, I would go for an explicit "enable reading these attributes".

dlech commented:
I updated this to add to new command line options.

I still think it is better to have reading debugfs attributes disabled by default to avoid the issues it can cause, so I've implemented the options this way.

There is an option to enable reading debugfs attributes just in case anyone was relying on that (although since it typically only reads register 0x00 if one even exists, I don't expect that it was actually very useful).

And there is a separate option to disable reading sysfs attributes just in case anyone wants to avoid poking hardware at all, but the default here is read and print the sysfs attribute values just like we have always done.

In all cases, the attribute names are still printed, it is just the values that may or may not be printed.

dNechita commented:
> Umm... I think the static checker needs work. The code compiles, so there clearly isn't a problem with the code itself.

You can ignore that finding.

dNechita commented on utils/iio_info.c:
```diff
@@ -33,12 +34,16 @@
 
 static const struct option options[] = {
-	{0, 0, 0, 0},
+	{ "read-debugfs", no_argument, NULL, 'd' },
```
When `iio_info` generates its output, it categorizes the attributes into context attributes, device attributes, buffer attributes, channel attributes, and debug attributes. Therefore, it makes sense to use the same naming. I suggest:
`{ "read-debug-attr", no_argument, NULL, 'd' }`

dNechita commented on utils/iio_info.c:
```diff
@@ -33,12 +34,16 @@
 static const char *options_descriptions[] = {
 	("[-x <xml_file>]\n"
 		"\t\t\t\t[-u <uri>]"),
+	("Read and print the value of debugfs attributes."),
+	("Do not read sysfs attributes, only print the names."),
```
I would rephrase these in order to:
 - use  'debug' instead of 'debugfs'
 - create more consistency between the 2 descriptions

I suggest:

`("Read and print the value of debug attributes."),`
`("Do not read the value of regular attributes."),`

dNechita commented:
When running with option `-n`, iio_info also hides the context attributes that could contain useful information such as the kernel version, the URI used to create the context, hardware version, hardware firmware version, etc. Additionally, these are not sysfs attributes. Instead, libiio reads them from an .ini file.
Therefore, it makes sense to always print their values. They can be easily detected by checking if they are of type:`IIO_ATTR_TYPE_CONTEXT`.

dlech commented:
Thanks for the suggestions, I have updated the changes accordingly.
